zulip_bots/zulip_bots/bots/idonethis/idonethis.py

```diff
@@ -0,0 +1,211 @@
+Either set a default team or pass the `--team` flag.
+More information in my help""")
+
+    team_id = get_team_hash(team)
+    data = api_create_entry(new_message, team_id)
+    return "Great work :thumbs_up:. New entry `{}` created!".format(data['body_formatted'])
+
+class IDoneThisHandler(object):
+    def initialize(self, bot_handler: Any) -> None:
+        global api_key, default_team
+        self.config_info = bot_handler.get_config_info('idonethis')
+        if 'key' in self.config_info:
+            api_key = self.config_info['key']
+        if 'default_team' in self.config_info:
+            default_team = self.config_info['default_team']
+        else:
+            logging.warn("Cannot find default team. Users will need to manually specify a team each time an entry is created.")
+
+        try:
+            api_noop()
+        except AuthenticationException:
+            logging.error("Authentication exception with idonethis. Can you check that your API keys are correct? ")
+            sys.exit(1)
+        except Exception:
+            logging.error("Problem connecting to idonethis.")
+            sys.exit(1)
+
+    def usage(self) -> str:
+        default_team_message = ""
+        if default_team:
+            default_team_message = "The default team is currently set as `" + default_team + "`."
+        else:
+            default_team_message = "There is currently no default team setup :frowning:."
+        return '''
+This bot allows for interaction with idonethis, a collaboration tool to increase a team's productivity.
+Below are some of the commands you can use, and what they do.
+
+`<team>` can either be the name or ID of a team.
+
+ * `@mention help` view this help message
+ * `@mention teams list` or `@mention list teams`
+    List all the teams
+ * `@mention team info <team>`
+    Show information about one `<team>`
+ * `@mention entries list` or `@mention list entries`
+    List entries from any team
+ * `@mention entries list <team>` or `@mention list entries <team>`
+    List all entries from `<team>`
+ * `@mention entries create` or `@mention new entry` or `@mention create entry` or `@mention new entry` or `@mention i did`
+    Create a new entry. Optionally supply `--team=<team>` for teams with no spaces or `"--team=<team>"`
+    for teams with spaces. For example `@mention i did "--team=product team" something` will create a
+    new entry `something` for the product team.
+        ''' + default_team_message
+
+    def handle_message(self, message: Any, bot_handler: Any) -> None:
+        bot_handler.send_reply(message, self.get_response(message))
+
+    def get_response(self, message: Any) -> str:
+        message_content = message['content'].strip().split()
+        if message_content == "":
+            return ""
+        reply = ""
+        try:
+            if " ".join(message_content[:2]) in ["teams list", "list teams"]:
```

Comment:
" ".join(message_content[:2]) is used a lot. You should assign it to a variable.